lib/hls/hls_parser.js

```diff
@@ -2935,6 +2935,12 @@ shaka.hls.HlsParser = class {
 
     if (!mimeType) {
       mimeType = await this.guessMimeType_(type, codecs, segments);
+
+      if (type == shaka.util.ManifestParserUtils.ContentType.TEXT &&
```

Comment:
I can't tell what this logic is doing.  Can you please add an explanatory comment?